### PR TITLE
updated models and analysis

### PR DESCRIPTION
**Pull request by 3mei**

### CODE REVIEW

**Comment by 3mei** [comment]
@jthandy I updated all models and analyses. Your comments are welcome!

File: models/zendesk_tickets.sql [hunk 1]
```diff
@@ -5,17 +5,16 @@
...
 select distinct
 	id as ticket_id,
 	status,
-	type,
-	via__channel as channel,
+	satisfaction_rating__score as satisfaction_rating,
 	created_at::datetime as ticket_date
-from zendesk_pipeline.tickets
+from zendesk.tickets
 where id in (select * from valid_tickets)
```
Comment (jthandy) [inline models/zendesk_tickets.sql:20]: large in clauses aren't performant--far better to inner join to the CTE; uses the optimizer better.

File: models/zendesk_audits_events.sql [hunk 2]
```diff
@@ -5,7 +5,7 @@ select
 	author_id,
 	_rjm_source_key_id as audit_id,
 	value
-from zendesk_pipeline.audits__events
+from zendesk.audits__events
 where
 	-- data for some tickets is missing. See zendesk_audits
 	_rjm_source_key_id in
```
Comment (jthandy) [inline models/zendesk_audits_events.sql:11]: large in clauses aren't performant--create a CTE and inner join it; uses the optimizer better.

Comment (jthandy) [inline models/zendesk_audits_events.sql:11]: same thing--create a CTE and inner join it

File: models/zendesk_ticket_summary.sql [hunk 3]
```diff
@@ -26,25 +33,25 @@ first_audit_dates as
         -- look at events from admins and agents only
         audit_author_id in (select * from admins_and_agents)
```
Comment (jthandy) [inline models/zendesk_ticket_summary.sql:34]: change to inner join

File: analysis/agent_stats.sql [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+
+author_stats as
+(
+	select
+		a.ticket_id, audit_author_id,
+		datediff(min, ticket_date, solved_date)/60.0 hours_to_solved,
+		satisfaction_score
+	from single_agent_tickets a
+	inner join {{ref("ac_yevgeniy.zendesk_ticket_summary")}} b
```
Comment (jthandy) [inline analysis/agent_stats.sql]: remove reference to schema

File: analysis/avg_med_hours_to_first_reply.sql [hunk 5]
```diff
@@ -3,11 +3,15 @@ reply_times as
 (
     select
         ticket_id, date_trunc('week',ticket_date)::date as ticket_week,
-        datediff(mins, ticket_date, first_reply_date)/60.0 hours_to_first_reply
-    from ac_yevgeniy.zendesk_ticket_summary
```
Comment (jthandy) [inline analysis/avg_med_hours_to_first_reply.sql]: use ref

File: analysis/avg_med_hours_to_solved.sql [hunk 6]
```diff
@@ -4,10 +4,14 @@ solved_times as
     select
         ticket_id, date_trunc('week',ticket_date)::date as ticket_week,
         datediff(min, ticket_date, solved_date)/60.0 hours_to_solved
-    from ac_yevgeniy.zendesk_ticket_summary
```
Comment (jthandy) [inline analysis/avg_med_hours_to_solved.sql]: use ref

File: analysis/agent_stats.sql [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+		on a.ticket_id = b.ticket_id
+	-- limit to solved tickets only
+	where solved_date is not null
+)
+
+select
+	audit_author_id,
+	avg(hours_to_solved) avg_hours_to_solved,
+	avg(satisfaction_score) as avg_satisfaction_score
```
Comment (jthandy) [inline analysis/agent_stats.sql:56]: would be great to add a quick count of tickets in here:
`count(ticket_id) num_tickets,`

File: analysis/new_solved_net_by_week.sql [hunk 8]
```diff
@@ -18,6 +21,8 @@ solved_tickets as
 (
 	select solved_week as week, count(*) as num_solved
 	from ticket_summary
+	-- make sure the ticket is actually solved
+	where solved_week is not null
 	group by solved_week
 )
```
Comment (jthandy) [inline analysis/new_solved_net_by_week.sql:28]: `cum_new` and `cum_solved` should actually just be new and solved. backlog should be cumulative, but the new and solved lines should be flows for that given week. just remove the window function for each of them.

**Comment by jthandy** [comment]
@3mei looks really solid. some line level comments but otherwise good to go. I've run everything in dbt and run all of the analyses in mode.

File: models/zendesk_tickets.sql (hunk 1 above)
Comment (3mei) [inline models/zendesk_tickets.sql:20]: @jthandy Is this actually true? I tested this out a while ago on our Linkedin dataset, which has much bigger tables, and subqueries performed much better than joins. Even in this example, the subquery performs a little faster. See also https://www.quora.com/Which-is-faster-joins-or-subqueries and http://www.chrislondon.co/joins-vs-subqueries/

File: analysis/agent_stats.sql (hunk 4 above)
Comment (3mei) [inline analysis/agent_stats.sql]: yep.

**Comment by jthandy** [comment]
@3mei I'm fine punting on the performance question for now. I don't want to get bogged down on it and our dataset isn't big enough to empirically test today. we can always modify in the future when it becomes an issue.

File: analysis/agent_stats.sql (hunk 7 above)
Comment (3mei) [inline analysis/agent_stats.sql:56]: done.

**Comment by jthandy** [comment]
I'm good! Merging...

**Pull request merged**